src/script/descriptor.cpp

```diff
@@ -841,6 +841,38 @@ class DescriptorImpl : public Descriptor
         return true;
     }
 
+    bool IsWatchOnly(const SigningProvider& arg) const override
+    {
+        std::set<CPubKey> pubkeys;
+        std::set<CExtPubKey> ext_pubkeys;
+        this->GetPubKeys(pubkeys, ext_pubkeys);
+        auto output_type{this->GetOutputType()};
+
+        size_t pubkey_count{pubkeys.size() + ext_pubkeys.size()};
+        size_t privkey_count{0};
+
+        if (pubkey_count == 0) {
+            return true;
+        }
+
+        for (auto pubkey : pubkeys) {
+            CKey key;
+            if (arg.GetKey(pubkey.GetID(), key)) {
+                privkey_count += 1;
+            } else if (output_type.has_value() && output_type == OutputType::BECH32M && arg.GetKeyByXOnly(XOnlyPubKey(pubkey), key)) {
```

Comment:
In 64213213b3691c9d93de95f880f820a640cd3c35 _descriptors: add IsWatchOnly()_: nit: you don't need `output_type.has_value()`